app/backend/core/authentication.py

```diff
@@ -241,12 +241,18 @@ async def get_auth_claims_if_enabled(self, headers: dict) -> dict[str, Any]:
     async def check_path_auth(self, path: str, auth_claims: dict[str, Any], search_client: SearchClient) -> bool:
         # Start with the standard security filter for all queries
         security_filter = self.build_security_filters(overrides={}, auth_claims=auth_claims)
-        # If there was no security filter, then the path is allowed
-        if not security_filter:
+        # If there was no security filter or no path, then the path is allowed
+        if not security_filter or len(path) == 0:
             return True
 
+        # Remove any fragment string from the path before checking
+        fragment_index = path.find("#")
```

Comment:
Is there a test for this new if?

Reply:
yes, it's in test_authenticationhelper.py